src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1050,13 +1050,14 @@ private function createService(Definition $definition, array &$inlineServices, b
         $parameterBag = $this->getParameterBag();
 
         if (true === $tryProxy && $definition->isLazy() && !$tryProxy = !($proxy = $this->proxyInstantiator ??= new LazyServiceInstantiator()) || $proxy instanceof RealServiceInstantiator) {
+            $containerRef = \WeakReference::create($this);
```

Comment:
should we recreate a new WeakReference or should we keep it in a (lazy-initialized) property of the ContainerBuilder ?

Reply:
good idea, updated